content/blog/espressif-devkits-with-wsl2/index.md

```diff
@@ -1,77 +1,103 @@
 ---
-title: "Working with Espressif's devkits using WSL2"
+title: "Working with Espressif's SoC's using WSL2"
 date: 2024-10-31
 showAuthor: false
-tags: ["ESP-IDF", "Windows", "WSL2"]
 authors:
-    - "jakub-kocka"
+    - jakub-kocka
+tags: ["ESP-IDF", "devkit", "Windows", "WSL", "WSL2", "USB"]
 ---
 
+Released in 2016 by Microsoft, the Windows Subsystem for Linux [(WSL)](https://learn.microsoft.com/en-us/windows/wsl/) is a feature that allows the use of a Linux environment with no need of a virtual machine, container or dual boot. The [WSL](https://learn.microsoft.com/en-us/windows/wsl/) is a great way to run Linux alongside Windows.
+
 ## Introduction
 
-[WSL](https://learn.microsoft.com/en-us/windows/wsl/) (Windows Subsystem for Linux) is a great way to run Linux alongside Windows. However, when there is a need to use it with Espressif's devkits, it can be a bit tricky due to the required serial port forwarding. In this short article, a step-by-step guide is provided to help you work with Espressif's devkits using WSL.
+Developing ESP32's based software solutions with Windows can be, sometimes, overwhelming due to some characteristics of the OS that increases the build time in some configurations or security polices.
+
+To overcome, the WSL is an good option to optimize, specially the build time. However, when there is a need to use it with Espressif's SoCs with USB-CDC or external USB-to-Serial (`CP210x`, etc), it can be a bit more tricky due to the required serial port forwarding.
```

Comment:
I am not sure if we should mention the build time here. For example, in my case using WSL is significantly (e.g. at least 5 times) slower than using a PowerShell console with an IDF environment.

This is also why I was indirect about the use case. We have seen these issues when users want to use SoCs with WSL, so this article exists, but to be honest, I am not certain about a clear and good use case.

Reply:
Good point! PTAL at the updated version, @jakub-kocka.

Reply:
Nice improvement, thank you @pedrominatel. LGMT!